scvi/core/models/base.py

```diff
@@ -179,23 +180,41 @@ def save(self, dir_path: str, overwrite: bool = False):
                     dir_path
                 )
             )
-        torch.save(self.model.state_dict(), os.path.join(dir_path, "model_params.pt"))
-        with open(os.path.join(dir_path, "attr.pkl"), "wb") as f:
+
+        if save_anndata:
+            self.adata.write(os.path.join(dir_path, "adata.h5ad"))
+
+        model_save_path = os.path.join(dir_path, "model_params.pt")
+        attr_save_path = os.path.join(dir_path, "attr.pkl")
+        varnames_save_path = os.path.join(dir_path, "var_names.pkl")
+
+        var_names = self.adata.var_names
```

Comment:
I think you should use the get_var_names_from_setup_anndata function because if using raw the names might be different. This will require some changes in load.

Reply:
Alternatively, should we just remove all use raw functionality? It seems like it's causing headaches for little value.

Reply:
most people use raw for the library size log normalized data (like in our tutorials), not the raw counts, and this is kind of the convention in scanpy.